cedar-policy/src/frontend/is_authorized.rs

```diff
@@ -324,88 +346,270 @@ fn constant_true() -> bool {
-        let schema = parse_schema(self.schema)?;
-        let principal = parse_entity_uid(self.principal, "principal")?;
-        let action = parse_action(self.action)?;
-        let resource = parse_entity_uid(self.resource, "resource")?;
-        let context = parse_context(self.context, schema.as_ref(), &action)?;
-        let q = Request::new(
+        let mut errs = vec![];
+        let mut warnings = vec![];
+        let schema = match self.schema.map(Schema::parse).transpose() {
+            Ok(None) => None,
+            Ok(Some((schema, new_warnings))) => {
+                warnings.extend(new_warnings);
+                Some(schema)
+            }
+            Err(e) => {
+                errs.push(e);
+                None
+            }
+        };
+        let principal = match self
+            .principal
+            .map(|p| parse_entity_uid(p, "principal"))
+            .transpose()
+        {
+            Ok(None) => None,
+            Ok(Some(euid)) => Some(euid),
+            Err(e) => {
+                errs.push(e);
+                None
+            }
+        };
+        let action = match parse_action(self.action) {
+            Ok(euid) => Some(euid),
+            Err(e) => {
+                errs.push(e);
+                None
+            }
+        };
+        let resource = match self
+            .resource
+            .map(|r| parse_entity_uid(r, "resource"))
+            .transpose()
+        {
+            Ok(None) => None,
+            Ok(Some(euid)) => Some(euid),
+            Err(e) => {
+                errs.push(e);
+                None
+            }
+        };
+        let context = match parse_context(self.context, schema.as_ref(), action.as_ref()) {
+            Ok(context) => Some(context),
+            Err(e) => {
+                errs.extend(e);
+                None
+            }
+        };
+        let q = match Request::new(
             principal,
-            Some(action),
+            action,
             resource,
-            context,
+            context.unwrap_or_else(|| Context::empty()),
             if self.enable_request_validation {
                 schema.as_ref()
             } else {
                 None
             },
         )
-        .map_err(|e| [e.to_string()])?;
-        let (policies, entities) = self.slice.try_into(schema.as_ref())?;
-        Ok((q, policies, entities))
+        .map_err(|e| e.to_string())
+        {
+            Ok(req) => Some(req),
+            Err(e) => {
+                errs.push(e);
+                None
+            }
+        };
+        let (policies, entities) = match self.slice.try_into(schema.as_ref()) {
+            Ok((policies, entities)) => (Some(policies), Some(entities)),
+            Err(e) => {
+                errs.extend(e);
+                (None, None)
+            }
+        };
+        if errs.is_empty() {
+            // TODO: this ignores warnings
```

Comment:
Do we want to adjust the APIs in #757 to allow returning non-fatal errors (i.e., warnings)?

Reply:
We probably should

Reply:
How about the following?
```rust
enum AuthorizationAnswer {
    Failure { errors: HashSet<serde_json::Value> },
    Success { 
        response: Response,
        warnings: HashSet<serde_json::Value>,
    },
}
```
The miette json format should support tagging an "error" with the severity "warning".

Reply:
Or, do we envision that `Response` (i.e., `Diagnostics`) will eventually evolve to include warnings?  Would it make more sense to put all warnings inside `Diagnostics`?

Reply:
Good question - I think that would also be reasonable.

Reply:
Tried the version with warnings in `Diagnostics` and updated this PR; let me know what you think.

Also accounted for the possibility that we have both errors and warnings, by adding a `warnings` field to `Failure` here as well.

Reply:
Having second thoughts.  In the fullness of time we probably want to support warnings generated by evaluating a particular policy, and also warnings generated when parsing/processing things like schemas and entity data.  The former are associated with a particular policy id and make sense to be in `Diagnostics`.  The latter are not and do not.  Perhaps we want a space for warnings both in `Diagnostics` (for warnings associated with evaluating a particular policy) and in `Failure` and `Success` generally.

Reply:
Adjusted again in 684e8f65 on this PR